File: snippets/python/src/foundations/actions/user_account.py
import restate
from pydantic import BaseModel
from restate import ObjectContext

class UserProfile(BaseModel):
    name: str
    email: str

user_account = restate.VirtualObject("UserAccount")


@user_account.handler()
async def get_profile(ctx: ObjectContext) -> UserProfile:
    return UserProfile(name="John", email="john@example.com")

File: snippets/java/src/main/java/foundations/actions/UserAccount.java
package foundations.actions;

import dev.restate.sdk.annotation.Handler;
import dev.restate.sdk.annotation.VirtualObject;

@VirtualObject
public class UserAccount {
  @Handler
  public UserProfile getProfile() {
    return new UserProfile("John", "john@example.com");
  }
}
